modules/apps/depot/depot-service/src/main/java/com/liferay/depot/internal/security/permission/contributor/DepotRoleContributor.java

```diff
@@ -85,6 +93,31 @@ private void _addRoleId(RoleCollection roleCollection, String roleName)
 		roleCollection.addRoleId(role.getRoleId());
 	}
 
+	private boolean _hasInheritedMemberships(long groupId, UserBag userBag) {
+		List<Organization> organizations =
+			_organizationLocalService.getGroupOrganizations(groupId);
+
+		for (Organization organization : organizations) {
+			if (userBag.hasUserOrg(organization)) {
+				return true;
+			}
+		}
+
+		List<UserGroup> groupUserGroups =
+			_userGroupLocalService.getGroupUserGroups(groupId);
+
+		List<UserGroup> userUserGroups =
+			_userGroupLocalService.getUserUserGroups(userBag.getUserId());
+
+		Stream<UserGroup> groupUserGroupsStream = groupUserGroups.stream();
+
+		if (groupUserGroupsStream.anyMatch(userUserGroups::contains)) {
+			return true;
+		}
```

Comment:
Stream usage is discouraged. Can you replace this with a regular for loop?

Reply:
ok :(